### ISSUE TITLE
visual.MovieStim() fails to return -1 on last movie frame on OSX

### ISSUE THREAD

**Issue by ejolly** [psychopy/psychopy#1495]
I can't seem to get `MovieStim()` to work properly on OSX 10.11.6, using avbin10. While the movie plays just fine, any code after the playback while loop (copied from the demo code) fails to run. From my own testing it seems like this happens because `mov.status` never returns anything other than 1, thus failing to break out of the loop which expects a -1 (i.e. value of `visual.FINISHED`). This does not however, happen with `MovieStim3()`.  

I'd prefer to use `MovieStim()` instead of `MovieStim3()` because I'm playing a rather long movie file and as noted in the [source](https://github.com/psychopy/psychopy/blob/master/psychopy/visual/movie3.py#L20), the audio file takes a long time to load up and my system memory usage increases quite a bit.

Here's some code I used to reproduce this. I've tested this with a bunch of different movies (e.g. mp4, mov, etc) encoded in different ways. Other programs, (e.g. quicktime, VLC), etc don't seem to have any problem playing these files, so I feel pretty confident it has nothing to do with the videos themselves. 

Any suggestions would be greatly appreciated! 

```
from psychopy import prefs
prefs.general['audioLib'] = ['pyo'] #tried switching this to pygame and that doesnt solve
from psychopy import visual

stim_path = 'path/to/my/movie.mp4'

window = visual.Window(size=(800,600), fullscr=False, screen=0, allowGUI=True, color='black') 
#doesn't seem to be a screen resolution/full screen issue either

mov = visual.MovieStim(window, stim_path, size=(800,600),
    flipVert=False, flipHoriz=False, loop=False)
    
while mov.status != visual.FINISHED:
    mov.draw()
    window.flip()
    print(mov.status) #Returns 1's even on the last frame; MovieStim3 returns -1 and thus breaks out of loop
    
window.close()
```

**Comment by peircej** [comment]
This might be a change in the pyglet handling of EOS events (see references to `on_eos` in psychopy/visual/movie.py), but I'm not personally motivated to dig around and work out why it's gone wrong. I want to work on MovieStim3 as the single solution for all the problems (i.e. fix it to use streaming audio for large movies) and leave the original MovieStim as an old unsupported option.

**Comment by peircej** [comment]
If you manage to fix the issue of on_eos being called then I'd happily take a fix, of course. In the meantime I think you'll need a workaround like detecting the movie duration and using that to decide whether or not the movie is still playing

**Comment by TEParsons** [comment]
I believe this is fixed by #6740
